operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java

```diff
@@ -644,27 +643,34 @@ private static void addUpdateRevisionAnnotation(String revision, StatefulSet toU
         toUpdate.getMetadata().getAnnotations().put(Constants.KEYCLOAK_UPDATE_REVISION_ANNOTATION, revision);
     }
 
-    record ManagementEndpoint(String relativePath, String protocol, int port) {}
+    record ManagementEndpoint(String relativePath, String protocol, int port, String portName) {}
 
-    static ManagementEndpoint managementEndpoint(Keycloak keycloakCR, Context<Keycloak> context, boolean useMgmtProtocolPort) {
+    static ManagementEndpoint managementEndpoint(Keycloak keycloakCR, Context<Keycloak> context, boolean health) {
         boolean tls = isTlsConfigured(keycloakCR);
         String protocol = tls ? "HTTPS" : "HTTP";
         int port;
+        String portName;
 
-        if (useMgmtProtocolPort) {
+        var legacy = readConfigurationValue(LEGACY_MANAGEMENT_ENABLED, keycloakCR, context).map(Boolean::valueOf).orElse(false);
+
+        var healthManagementEnabled = readConfigurationValue(CRDUtils.HTTP_MANAGEMENT_HEALTH_ENABLED, keycloakCR, context).map(Boolean::valueOf).orElse(true);
+
+        if (!legacy && (!health || healthManagementEnabled)) {
             port = HttpManagementSpec.managementPort(keycloakCR);
+            portName = Constants.KEYCLOAK_MANAGEMENT_PORT_NAME;
             if (readConfigurationValue(HTTP_MANAGEMENT_SCHEME, keycloakCR, context).filter("http"::equals).isPresent()) {
                 protocol = "HTTP";
             }
         } else {
             port = tls ? HttpSpec.httpsPort(keycloakCR) : HttpSpec.httpPort(keycloakCR);
+            portName = tls ? Constants.KEYCLOAK_HTTP_PORT_NAME : Constants.KEYCLOAK_HTTP_PORT_NAME;
```

Comment:
Identical branches, do you mean:

`portName = tls ? Constants.KEYCLOAK_HTTPS_PORT_NAME : Constants.KEYCLOAK_HTTP_PORT_NAME;` ?